torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -367,8 +368,10 @@ void ProcessGroupNCCL::WorkNCCL::synchronizeInternal(
         // if throwing timed out excepiton without aborting nccl communicators
         // here, it was observed that CUDA GPU will have 100% utilization and
         // can not run new events successfully.
+        std::stringstream ss;
+        ss << "Work: " << *this << " timed out in call to wait().";
         for (const auto& ncclComm : ncclComms_) {
-          ncclComm->ncclCommAbort();
+          ncclComm->ncclCommAbort(ss.str());
```

Comment:
Don't we need to check for `commFailureReason_` to be set on line 388 and appropriately return that error instead of the time out?

Also, shouldn't we update `checkAndSetException` to first check for `commFailureReason_` first and use that exception instead?

Reply:
This part of the code is for NCCL_BLOCKING_WAIT and we are in a section where the work has timed out, so isn't the root cause of aborting the communicator a timeout and we should pass that reason into `ncclCommAbort`? The goal here is to basically make `ncclCommAbort` aware of why we are aborting it and in this case it is a work timeout.

> Also, shouldn't we update checkAndSetException to first check for commFailureReason_ first and use that exception instead?

Yes, looks like we should do this.

Reply:
Yep, `ProcessGroupNCCL::WorkNCCL::checkAndSetException()` will try to grab the `commFailureReason` from the NCCL communicator if it exists, via `checkForNcclErrorsInternal`.